iree/compiler/Conversion/Common/PadLinalgOps.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "mlir/Dialect/Linalg/IR/LinalgOps.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+/// A pattern to pad staticly shaped matmul operands to the next integer
+/// multiple of padSize.
+class PadMatmulOp : public OpRewritePattern<linalg::MatmulOp> {
+ public:
+  PadMatmulOp(MLIRContext *context, int size, PatternBenefit benefit = 1)
+      : OpRewritePattern<linalg::MatmulOp>(context, benefit),
+        paddingSize(size) {}
+
+  LogicalResult matchAndRewrite(linalg::MatmulOp matmulOp,
+                                PatternRewriter &rewriter) const override {
+    auto loc = matmulOp.getLoc();
+    auto lhs = matmulOp.inputs()[0];
+    auto rhs = matmulOp.inputs()[1];
+    auto result = matmulOp.outputs()[0];
+
+    if (lhs.getDefiningOp<linalg::PadTensorOp>() ||
+        rhs.getDefiningOp<linalg::PadTensorOp>()) {
+      return failure();
+    }
+
+    auto lhsShapeType = lhs.getType().cast<ShapedType>();
+    auto rhsShapeType = rhs.getType().cast<ShapedType>();
+
+    if (!lhsShapeType || !rhsShapeType) return failure();
+
+    if (!lhsShapeType.hasStaticShape() || !rhsShapeType.hasStaticShape()) {
+      return failure();
+    }
+
+    auto lhsShape = lhsShapeType.getShape();
+    auto rhsShape = rhsShapeType.getShape();
+
+    int M = lhsShape[0], K = lhsShape[1], N = rhsShape[1];
+
+    int newMSize = std::ceil(float(M) / paddingSize) * paddingSize;
+    int newNSize = std::ceil(float(N) / paddingSize) * paddingSize;
+    int newKSize = std::ceil(float(K) / paddingSize) * paddingSize;
+
+    int paddingForM = newMSize - M;
+    int paddingForN = newNSize - N;
+    int paddingForK = newKSize - K;
+
+    if (paddingForM == 0 && paddingForN == 0 && paddingForK == 0)
+      return failure();
+
+    auto getPaddedOperand = [&](Value operand, ArrayRef<int64_t> shape,
```

Comment:
Instead of this lambda, maybe you can just use this method 
https://github.com/llvm/llvm-project/blob/cef9b96b01b75fedea5e91ece776228f4088ba78/mlir/include/mlir/Dialect/Linalg/IR/LinalgOps.td#L241

Reply:
Thanks!